### PR TITLE
[NA] Move Codegen Docs back

### PR DESCRIPTION
**Pull request by cipolleschi**
This PR moves the Codegen docs back into the website, in preparation of the New Architecture by default in 0.76.

### CODE REVIEW

File: docs/the-new-architecture/codegen-cli.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# The Codegen CLI
+
+Calling gradle or manually calling a script might be hard to remember and it requeires a lot of cerimonies.
```
Comment (blakef) [inline docs/the-new-architecture/codegen-cli.md:3]:
```suggestion
Calling Gradle or manually calling a script might be hard to remember and it requires a lot of ceremony.
```

File: docs/the-new-architecture/codegen-cli.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+To simplify it, we created a CLI tool that can help you running those tasks: the **Codegen** cli.
+
+```shell
+npx react-native codegen [--path path] [--platform string] [--outputPath path]
+```
+
+This command runs `react-native-codegen` for your project. The following options are available:
```
Comment (blakef) [inline docs/the-new-architecture/codegen-cli.md:11]:
```suggestion
This command runs [react-native-codegen](https://www.npmjs.com/package/react-native-codegen) for your project. The following options are available:
```

File: docs/the-new-architecture/codegen-cli.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+- `--path` - Path to `package.json`. The default path is the current working directory.
+- `--platform` - Target platform. Supported values: `android`, `ios`, `all`. The default value is all.
```
Comment (blakef) [inline docs/the-new-architecture/codegen-cli.md:14]:
```suggestion
- `--platform` - Target platform. Supported values: `android`, `ios`, `all`. The default value is `all`.
```

File: docs/the-new-architecture/codegen-cli.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+- `--outputPath` - Output path. The default value is the value defined in `codegenCofig.outputDir`.
```
Comment (blakef) [inline docs/the-new-architecture/codegen-cli.md:15]:
```suggestion
- `--outputPath` - Output path. The default value is the value defined in `codegenConfig.outputDir`.
```

File: docs/the-new-architecture/codegen-cli.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+- Since all native code is there, it is possible to ship the native part of the library as a prebuild.
+
+To enable this setup:
+
+- Add the `includesGeneratedCode` property into your library's `codegenConfig` field in the `package.json` file. Set its value to `true`.
+- Run **Codegen** locally with the codegen CLI.
+- Update your `package.json` to include the generated code.
+- Update your `podspec` to include the generated code.
+- Update your `build.gradle` file to include the generated code.
+- Update `cmakeListsPath` in `react-native.config.js` (example) so that Gradle doesn't look for CMakeLists file in the build directory but instead in your outputDir.
```
Comment (blakef) [inline docs/the-new-architecture/codegen-cli.md:59]: What does `(example)` mean?  Is this a note for later work?

File: docs/the-new-architecture/using-codegen.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
+# Using Codegen
+
+This guide teaches how to configure **Codegen**, and how to invoke it manually for each platform, and describes the generated code.
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:3]:
```suggestion
This guide teaches how to:
- configure **Codegen**,
- invoke it manually for each platform, and
- describes the generated code.
```

File: docs/the-new-architecture/using-codegen.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+For the sake of this guide, create a project using the React Native CLI as follows:
+
+```bash
+npx @react-native-community/cli@latest init SampleApp --version 0.76.0
+```
+
+**Codegen** is used to generate the glue-code for your custom modules or components. See the guides for Turbo Native Modules and Fabric Native Components for more details on how to create them.
+
+<!-- TODO: add links -->
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:19]: 👍

File: docs/the-new-architecture/using-codegen.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+- `android.javaPackageName`: this is an android specific setting to let **Codegen** generate the files in a custom package.
+
+When **Codegen** runs, it searches among all the dependencies of the app, looking for JS files that respects some specific conventions, and it generates the required code:
+
+- Turbo Native Modules requires that the spec files are prefixed with `Native`. For example, `NativeLocalStorage.ts` is a valid name for a spec file.
+- Native Fabric Components requires that the spec files are suffixed with `NativeComponent`. For example, `WebViewNativeComponent.tx` is a valid name for a spec file.
+
+## Running **Codegen**
+
+The rest of this guide assumes that you have a Native Turbo Module and/or a Native Fabric Component already set up in your project. We also assume that you have valid specification files in the `jsSrcsDir` specified in the `package.json`.
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:53]:
```suggestion
The rest of this guide assumes that you have a Native Turbo Module, Native Fabric Component or both already set up in your project. We also assume that you have valid specification files in the `jsSrcsDir` specified in the `package.json`.
```
Strangely this comes up with Shrunk & White and the suggestion is to do the above.

File: docs/the-new-architecture/using-codegen.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+### Android
+
+**Codegen** for Android is integrated with the React Native Gradle Plugin (RNGP). The RNGP contains a task that can be invoked that reads the configurations defined in the `package.json` file and execute **Codegen**. To run the gradle task, first navigate inside the `android `folder of your project. Then run:
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:57]:
```suggestion
**Codegen** for Android is integrated with the React Native Gradle Plugin (RNGP). The RNGP contains a task that can be invoked that reads the configurations defined in the `package.json` file and execute **Codegen**. To run the Gradle task, first navigate inside the `android `folder of your project. Then run:
```

File: docs/the-new-architecture/using-codegen.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+```bash
+./gradlew generateCodegenArtifactsFromSchema
+```
+
+This task invokes the `generateCodegenArtifactsFromSchema` command on all the the imported projects of the app (the app and all the node modules which are linked to it). It generates the code in the corresponding `node_modules/<dependency>` folder. So, for example, if you have a Fabric Native Component whose node module is called `my-fabric-component`, the generated code is located in the `SampleApp/node_modules/my-fabric-component/android/build/generated/source/codegen` path. For the app, the code is generated in the `android/app/build/generated/source/codegen` folder.
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:63]: I do this a lot (I'm going to stop adding notes about it): 
- we should avoid starting a sentence with 'So'.  It doesn't add anything.
- be careful when using proper nouns: `gradle` vs "Gradle does X", `node` vs "whose Node module".
- avoid using "should", "would", "could" unless there is real doubt (stealing again from Shrunk & White).

File: docs/the-new-architecture/using-codegen.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+    --targetPlatform <android | ios>
+```
+
+where:
+
+- `--path` is the path to the root folder of your app.
+- `--outputPath` is the destination where **Codegen** should write the generated files.
+- `--targetPlatform` is the platform you'd like to generate the code for.
+
+#### The Generated Code
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:145]: I'd change this section to:

```markdown
### Example

Running the script with these arguments:
...
Will generate these files in the `ios/build` folder:
...
```

File: docs/the-new-architecture/using-codegen.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+Assuming that you ran the script with the command:
+
+```shell
+node node_modules/react-native/scripts/generate-Codegen-artifacts.js \
+    --path . \
+    --outputPath ios/ \
+    --targetPlatform ios
+```
+
+You should now find a new `build` folder in your `ios` app. The foulder should look something like this:
```
Comment (blakef) [inline docs/the-new-architecture/using-codegen.md:156]:
```suggestion
You should now find a new `build` folder in your `ios` app. The folder should look something like this:
```

**Review by blakef — APPROVED** [review]
Thank you for contributing so much towards the docs, this is great.  Almost all minor nits.  Approving to unblock you.

File: docs/the-new-architecture/codegen-cli.md (hunk 5 above)
Comment (cipolleschi) [reply docs/the-new-architecture/codegen-cli.md:59]: I forgot to remove it! xD
In the original docs there were links to examples, but t think they are valid anymore.

**Review by blakef — APPROVED** [review]

File: docs/the-new-architecture/codegen-cli.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
+# The Codegen CLI
+
+Calling Gradle or manually calling a script might be hard to remember and it requires a lot of cerimony.
+
+To simplify it, we created a CLI tool that can help you running those tasks: the **Codegen** cli.
+
+```shell
+npx react-native codegen [--path path] [--platform string] [--outputPath path]
```
[the remark this answers is not in the document]
Comment (cipolleschi) [reply docs/the-new-architecture/codegen-cli.md:8]: this actually works in 0.76. I tested it before writing the paragraph.

**Pull request merged**